Sources/Networking/Modifiers/Interceptors/Authorization/AuthorizationManaging.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AuthorizationManaging.swift
+//  
+//
+//  Created by Dominika Gajdová on 20.12.2022.
+//
+
+import Foundation
+
+/// AuthorizationManaging authorizes requests and manages refresh token mechanism
+/// AuthorizationStorageManaging is required to read & store `AuthorizationData`
+public protocol AuthorizationManaging {
+    var storage: any AuthorizationStorageManaging { get }
+    
+    func refreshAuthorizationData(with refreshToken: String) async throws -> AuthorizationData
+    func authorizeRequest(_ request: URLRequest) async throws -> URLRequest
+}
+
+public extension AuthorizationManaging {
+    func authorizeRequest(_ request: URLRequest) async throws -> URLRequest {
+        let accessToken = try await getValidAccessToken()
+        
+        /// Append authentication header to request and return it.
+        var mutableRequest = request
+        mutableRequest.setValue(
+            "Bearer \(accessToken)",
+            forHTTPHeaderField: HTTPHeader.HeaderField.authorization.rawValue
+        )
+        return mutableRequest
+    }
+    
+    func refreshAuthorizationData() async throws {
+        let refreshToken = try await storage.get().refreshToken
+        let newAuthData = try await refreshAuthorizationData(with: refreshToken)
+        
+        try await storage.save(data: newAuthData)
+    }
+    
+    func getValidAccessToken() async throws -> String {
```

Comment:
Should this also be in a protocol definition?

Reply:
Sure, I think we can add it there in case someone would want a custom implementation of this function 👍